### ISSUE TITLE
no-mergeable-namespace: Allow separate declarations for `N.A` and `N.B`

### ISSUE THREAD

**Issue by andy-ms** [palantir/tslint#3082]
### Bug Report

- __TSLint version__: 5.5.0
- __TypeScript version__: 2.5.0-dev.20170727
- __Running TSLint via__: CLI

#### TypeScript code being linted

```ts
namespace N.A {}
namespace N.B {}
```

with `tslint.json` configuration:

```json
{
    "rules": {
        "no-mergeable-namespace": true
    }
}
```

#### Actual behavior

Gives an error asking to merge the namespaces.

#### Expected behavior

Don't ask to do this, because they can't be fully merged, and some might prefer this style over nesting `namespace N { namespace A { } namespace B {} }`.

**Comment by JoshuaKGoldberg** [comment]
☠️ *TSLint's time has come!* ☠️ 

TSLint is no longer accepting most feature requests per #4534. See **[typescript-eslint.io](https://typescript-eslint.io)** for the new, shiny way to lint your TypeScript code with **ESLint**. ✨ 

It was a pleasure open sourcing with you all!

**Comment by JoshuaKGoldberg** [comment]
🤖 Beep boop! 👉  **TSLint is deprecated** 👈 _(#4534)_  and you should **switch to [typescript-eslint](https://typescript-eslint.io)**! 🤖 

🔒 This issue is being locked to prevent further unnecessary discussions. Thank you! 👋
